extension/lib/bigcommerce/JwtFactory.js

```diff
@@ -0,0 +1,59 @@
+const jwt = require('jwt-simple')
+const randomBytes = require('randombytes')
+
+/**
+ * @property {string} _clientId
+ * @property {string} _storeHash
+ * @property {string} _clientSecret
+ */
+class JwtFactory {
+  /**
+   * @param {string} clientId
+   * @param {string} storeHash
+   * @param {string} clientSecret
+   */
+  constructor (clientId, storeHash, clientSecret) {
+    this._clientId = clientId
+    this._storeHash = storeHash
+    this._clientSecret = clientSecret
+  }
+
+  /**
+   * @param {string} customerId
+   * @param {string|null} redirectLink
+   */
+  create (customerId, redirectLink = null) {
+    const payload = {
+      iss: this._clientId,
+      iat: Math.round((new Date()).getTime() / 1000),
+      jti: this._bin2hex(randomBytes(32)),
```

Comment:
There is already available conversion via toString('hex') method, it produces completely different output for the same bytes. I'd prefer to use this one - please try it instead.